src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSKmerUtils.java

```diff
@@ -0,0 +1,176 @@
+
+        //Initialize progress state
+        final long totalBases = records.values().stream().mapToLong(refBases -> refBases.getBases().length).sum();
+        logger.info("Generating kmers from " + totalBases + " bases in " + records.size() + " records...");
+        final ProgressCounter counter = new ProgressCounter(totalBases, 1e6, "Mbp", logger);
+
+        //Get kmers from each reference record
+        for (final String recName : records.keySet()) {
+
+            //Kmerize the record
+            final byte[] bases = records.get(recName).getBases();
+            final long[] list = SVKmerizer.stream(bases, kSize, kSpace, new SVKmerShort(kSize))
+                    .mapToLong(kmer -> new Long(PSKmerCollection.canonicalizeAndMask((SVKmerShort) kmer, kSize, mask)))
+                    .toArray();
+
+            //Add kmers to the result
+            collection.add(list);
+
+            counter.update(bases.length);
+        }
+        logger.info("Finished generating kmers!");
+        return collection;
+    }
+
+    /**
+     * Returns the total number of longs in the Collection
+     */
+    public static long longArrayCollectionSize(final Collection<long[]> lists) {
+        return lists.stream().mapToLong(arr -> arr.length).sum();
+    }
+
+    /**
+     * Converts a Collection of Lists of Longs's into a Hopscotch set
+     */
+    protected static LargeLongHopscotchSet longArrayCollectionToSet(final Collection<long[]> longs, final long numLongs) {
+        final LargeLongHopscotchSet kmerHopscotchSet = new LargeLongHopscotchSet(numLongs);
+        for (final long[] array : longs) {
+            for (final long val : array) {
+                kmerHopscotchSet.add(val);
+            }
+        }
+        return kmerHopscotchSet;
+    }
+
+    /**
+     * Converts a Collection of Lists of Longs's into a Bloom filter
+     */
+    protected static LongBloomFilter longArrayCollectionToBloomFilter(final Collection<long[]> longs, final long numLongs, final double bloomFpp) {
+        final LongBloomFilter bloomFilter = new LongBloomFilter(numLongs, bloomFpp);
+        final ProgressCounter counter = new ProgressCounter(numLongs, 1e6, "million kmers", logger);
+        for (final long[] array : longs) {
+            bloomFilter.addAll(array);
+            counter.update(array.length);
+        }
+        return bloomFilter;
+    }
+
+    /**
+     * Writes an object to a URI using Kryo serialization.
+     */
+    public static void writeKryoObject(final Object obj, String uri) {
+        final Output output = new Output(BucketUtils.createFile(uri));
+        final Kryo kryo = new Kryo();
+        kryo.writeObject(output, obj);
+        output.close();
+    }
+
+    public static void writeKmerSet(final String uri, final PSKmerSet set) {
+        String filePath = uri;
+        if (HOPSCOTCH_SET_EXTENSION != null && !uri.toLowerCase().endsWith(HOPSCOTCH_SET_EXTENSION.toLowerCase())) {
```

Comment:
I suppose I was being overly-cautious here. I'll compromise and just assume it's non-null at least. What I really need is endsWithIgnoreCase().